src/wwwroot/controllers/TemplateCtrl.js

```diff
@@ -66,7 +69,11 @@
         id: $scope.template.id
       }).then(function (result) {
         $scope.template.id = result;
-        $location.path('/templates');
+        if(url == "mseditor") {
+          location.href = location.origin + '/template-editor/?idCampaign=' + result;
+        } else {
+          $location.path('/templates');
+        }
```

Comment:
I do not like these conditions, `url` name is not clear. What do you think about keeping `save` method private and add another two methods for the different behaviors? call me if you need more details.